pkg/tanka/find.go

```diff
@@ -45,7 +45,11 @@ func FindEnvs(path string, opts FindOpts) ([]*v1alpha1.Environment, error) {
 func find(path string) ([]*v1alpha1.Environment, error) {
 	// try if this has envs
 	list, err := List(path, Opts{})
-	if len(list) != 0 && err == nil {
```

Comment:
This error is actually ignored on purpose. 

Consider `tk env list`: You run it against any directory that usually is not an environment. By checking the error you would fail here, because no `main.jsonnet` cannot be found (as expected). 

Instead you want to recurse `find` until you reach something where `List` returns no error.

Why are you interested in this error? When do you need it?

Reply:
In this case `tk export` fails silently:
```
// main.jsonnet
{
  namespace: error '',
  env: {
    apiVersion: 'tanka.dev/v1alpha1',
    kind: 'Environment',
    metadata: {
      name: 'environments/default',
    },
    spec: {
      apiServer: '',
      namespace: $.abc,
      resourceDefaults: {},
      expectVersions: {},
    },
    data: {},
  },
}
```

`tk export -r outputDir main.jsonnet`

FindEnvs only returns valid environments but also considers jsonnet errors as non-valid environments and hides them.

Reply:
I guess we need more granular errors.

Reply:
Something like this?
```go
	// try if this has envs
	list, err := List(path, Opts{})
	if len(list) != 0 && err == nil {
		// it has. don't search deeper
		return list, nil
	}
	switch err.(type) {
	case jpath.ErrorFileNotFound:
		break
	case jpath.ErrorNoBase:
		break
	default:
		return nil, err
	}
```

Reply:
I agree, swallowing Jsonnet errors is not a good practice.

We cannot really fail early, because we also use `FindEnvs` for generating the suggestions `tk apply`, etc have on the cli and only because a single environment fails you should not suddenly get no completions.

What about collecting all errors that occur and ultimately return them? Consumers of the function could then decide to handle the case (`tk export`) or ignore it (CLI completion)